modules/031-linstor/images/linstor-pools-importer/main.go

```diff
@@ -0,0 +1,594 @@
+		}
+		lvName, vgName, lvAttr, uuid, tags := strings.TrimSpace(a[0]), a[1], a[2], a[3], strings.Split(a[4], ",")
+		if lvName == "" {
+			return nil, fmt.Errorf("LV name can't be empty (line: %q)", line)
+		}
+		if vgName == "" {
+			return nil, fmt.Errorf("vgName can't be empty (line: %q)", line)
+		}
+		if lvAttr == "" {
+			return nil, fmt.Errorf("lvAttr can't be empty (line: %q)", line)
+		}
+		if uuid == "" {
+			return nil, fmt.Errorf("uuid can't be empty (line: %q)", line)
+		}
+		name, err := parseNameFromLVMTags(&tags)
+		switch {
+		case lvAttr[0:1] != "t":
+			skipReason = "is not a thin pool"
+		case err != nil:
+			skipReason = "has no propper tag set: " + err.Error()
+		}
+
+		sps = append(sps, Candidate{
+			Name:       "LVM Logical Volume " + vgName + "/" + lvName,
+			UUID:       uuid,
+			SkipReason: skipReason,
+			StoragePool: lclient.StoragePool{
+				StoragePoolName: name,
+				NodeName:        nodeName,
+				ProviderKind:    lclient.LVM_THIN,
+				Props: map[string]string{
+					"StorDriver/LvmVg":    vgName,
+					"StorDriver/ThinPool": lvName,
+				},
+			}})
+	}
+	return sps, nil
+}
+
+func parseLVMVolumeGroups(nodeName, out string) ([]Candidate, error) {
+	var sps []Candidate
+	for _, line := range strings.Split(out, "\n") {
+		var skipReason string
+		// Parsing line
+		if line == "" {
+			continue
+		}
+		a := strings.Split(line, ";")
+		if len(a) != 3 {
+			return nil, fmt.Errorf("wrong line: %q", line)
+		}
+		vgName, uuid, tags := strings.TrimSpace(a[0]), a[1], strings.Split(a[2], ",")
+		if vgName == "" {
+			return nil, fmt.Errorf("VG name can't be empty (line: %q)", line)
+		}
+		if uuid == "" {
+			return nil, fmt.Errorf("uuid can't be empty (line: %q)", line)
+		}
+		name, err := parseNameFromLVMTags(&tags)
+		if err != nil {
+			skipReason = "has no propper tag set: " + err.Error()
+		}
+
+		sps = append(sps, Candidate{
+			Name:       "LVM Volume Group " + vgName,
+			UUID:       uuid,
+			SkipReason: skipReason,
+			StoragePool: lclient.StoragePool{
+				StoragePoolName: name,
+				NodeName:        nodeName,
+				ProviderKind:    lclient.LVM,
+				Props: map[string]string{
+					"StorDriver/LvmVg": vgName,
+				},
+			},
+		})
+	}
+	return sps, nil
+}
+
+func parseNameFromLVMTags(tags *[]string) (string, error) {
+	var foundNames []string
+	for _, tag := range *tags {
+		t := strings.Split(tag, "-")
+		if t[0] == linstorPrefix && t[1] != "" {
+			foundNames = append(foundNames, strings.TrimPrefix(tag, linstorPrefix+"-"))
+		}
+	}
+	switch len(foundNames) {
+	case 0:
+		return "", errors.New("can't find tag with prefix " + linstorPrefix)
+	case 1:
+		return foundNames[0], nil
+	default:
+		return "", errors.New("found more than one tag with prefix " + linstorPrefix)
+	}
+}
+
+func newKubernetesEvent(nodeName string, owner *v1.ObjectReference, eventType, reason, message string) v1.Event {
+	eventTime := metav1.Now()
+	event := v1.Event{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace:    owner.Namespace,
```

Comment:
If we create storage classes, events should be global-scoped, as far as I see it.

Reply:
Not sure if this is possible, as the event resource is namespaced:

```
NAME                                  SHORTNAMES                             APIVERSION                             NAMESPACED   KIND
events                                ev                                     v1                                     true         Event
events                                ev                                     events.k8s.io/v1                       true         Event
```

Reply:
This is definitely possible. You have events for PVs, Nodes, and other global resources. They should be stored in the default namespace.

Reply:
Real-world example of the persistent volume event:
```
- apiVersion: v1
  count: 456
  eventTime: null
  firstTimestamp: "2022-04-05T09:49:24Z"
  involvedObject:
    apiVersion: v1
    kind: PersistentVolume
    name: pvc-fe7e237c-5dfc-4233-b904-2c23e9d0d599
    resourceVersion: "378666361"
    uid: d92ec83a-6cd1-4d77-842f-473d632a2f5a
  kind: Event
  lastTimestamp: "2022-04-08T12:52:57Z"
  message: persistentvolume pvc-fe7e237c-5dfc-4233-b904-2c23e9d0d599 is still attached
    to node standard-system-90d15d73-7495c4dccf-69hqr
  metadata:
    creationTimestamp: "2022-04-06T18:03:07Z"
    name: pvc-fe7e237c-5dfc-4233-b904-2c23e9d0d599.16e2f727db82a6c5
    namespace: default
    resourceVersion: "496509493"
    uid: 42484d59-9260-42f7-b089-97c1d8b82ac3
  reason: VolumeFailedDelete
  reportingComponent: ""
  reportingInstance: ""
  source:
    component: cinder.csi.openstack.org_standard-master-0_496ca7ac-eb09-4913-b7e6-87af23403f52
  type: Warning
```

Reply:
Okay I decided to use Node object as involvedObject.
This will generate global events in default namespace.